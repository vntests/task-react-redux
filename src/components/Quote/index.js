import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import { Form, FormGroup, Label, Input, Button, Modal, ModalHeader, ModalBody } from 'reactstrap';

import './index.css';

export default class Quote extends Component {
  static propTypes = {
    title: PropTypes.string,
    content: PropTypes.string,
    link: PropTypes.string,
    update: PropTypes.func,
    remove: PropTypes.func
  };

  constructor(props) {
    super(props);

    this.state = {
      title: this.props.title,
      content: this.props.content,
      link: this.props.link,
      editQuoteModal: false
    };
  }

  toggleEditQuoteModal() {
    this.setState({
      editQuoteModal: !this.state.editQuoteModal
    });
  }

  onChangeEditQuoteTitle(e) {
    this.setState({
      title: e.target.value
    });
  }

  onChangeEditQuoteContent(e) {
    this.setState({
      content: e.target.value
    });
  }

  onChangeEditQuoteLink(e) {
    this.setState({
      link: e.target.value
    });
  }

  onSubmitEditQuote(e) {
    e.preventDefault();

    const { id, update } = this.props;

    this.toggleEditQuoteModal();

    update({
      id,
      title: this.state.title,
      content: this.state.content,
      link: this.state.link
    });
  }

  render() {
    const {
      title,
      content,
      link,
      remove
    } = this.props;

    return (
      <Fragment>
        <blockquote className="blockquote quote">
          <Button
            type="button"
            size="sm"
            color="primary"
            onClick={() => { this.toggleEditQuoteModal() }}
          >
            Edit
          </Button>
          <Button
            type="button"
            size="sm"
            color="danger"
            onClick={remove}
          >
            Remove
          </Button>
          <div dangerouslySetInnerHTML={{__html: content}} />
          <footer className="blockquote-footer">
            {title} (<a href={link}>{link}</a>)
          </footer>
        </blockquote>

        <Modal
          isOpen={this.state.editQuoteModal}
          toggle={() => { this.toggleEditQuoteModal() }}
        >
          <ModalHeader
            toggle={() => { this.toggleEditQuoteModal() }}
          >
            Edit
          </ModalHeader>
          <ModalBody>
            <Form
              onSubmit={e => { this.onSubmitEditQuote(e) }}
            >
              <FormGroup>
                <Label for="editQuoteTitle">
                  Title
                </Label>
                <Input
                  type="text"
                  value={this.state.title}
                  name="editQuoteTitle"
                  id="editQuoteTitle"
                  onChange={e => { this.onChangeEditQuoteTitle(e) }}
                />
              </FormGroup>
              <FormGroup>
                <Label for="editQuoteContent">
                  Content
                </Label>
                <Input
                  type="textarea"
                  value={this.state.content}
                  name="editQuoteContent"
                  id="editQuoteContent"
                  onChange={e => { this.onChangeEditQuoteContent(e) }}
                />
              </FormGroup>
              <FormGroup>
                <Label for="editQuoteLink">
                  Link
                </Label>
                <Input
                  type="text"
                  value={this.state.link}
                  name="editQuoteLink"
                  id="editQuoteLink"
                  onChange={e => { this.onChangeEditQuoteLink(e) }}
                />
              </FormGroup>
              <Button
                type="submit"
                block
                color="primary"
              >
                Submit
              </Button>
            </Form>
          </ModalBody>
        </Modal>
      </Fragment>
    );
  }
};
