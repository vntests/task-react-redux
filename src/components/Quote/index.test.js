import React from 'react';
import renderer from 'react-test-renderer';

import Quote from './';

test('<Quote /> should render', () => {
  const component = renderer.create(
    <Quote />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('<Quote /> should render quote with title "Title", content "Content", link "Link"', () => {
  const component = renderer.create(
    <Quote
      title="Title"
      content="Content"
      link="Link"
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
