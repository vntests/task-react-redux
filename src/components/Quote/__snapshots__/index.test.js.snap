// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quote /> should render 1`] = `
<blockquote
  className="blockquote quote"
>
  <button
    className="btn btn-primary btn-sm"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>
  <button
    className="btn btn-danger btn-sm"
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
  <footer
    className="blockquote-footer"
  >
     (
    <a
      href={undefined}
    />
    )
  </footer>
</blockquote>
`;

exports[`<Quote /> should render quote with title "Title", content "Content", link "Link" 1`] = `
<blockquote
  className="blockquote quote"
>
  <button
    className="btn btn-primary btn-sm"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>
  <button
    className="btn btn-danger btn-sm"
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "Content",
      }
    }
  />
  <footer
    className="blockquote-footer"
  >
    Title
     (
    <a
      href="Link"
    >
      Link
    </a>
    )
  </footer>
</blockquote>
`;
